README.md

```diff
@@ -12,194 +12,129 @@ shared. These resources have a different lifecycle than resources in workspaces.
 
-Landing zones are implemented using the factory pattern; the factory creates *Landing Zone Definitions* (LZDs).
+Below are lists of changes required to introduce new landing zone:
+1) Implement specific steps which are responsible for Azure resource creation (see example below for `CreateVnetStep`)
+2) Implement specific steps' definition provider (by implementing `StepsDefinitionProvider` interface)
+3) Introduce new landing zone type by extending `StepsDefinitionFactoryType`
+4) Update mapping at `LandingZoneStepsDefinitionProviderFactory` based on landing zone type from section 3. This mapping is used by sub-flight to get flights' steps to create specific Azure resources.
 
-Landing Zone Definitions are where resources and their purpose are defined.
+In case of updating existing landing zone it is required to do following:
+1) Introduce new or adjusting existing step
+2) In case of a new step it is also required to include it into corresponding steps' provider
 
-A Landing Zone Definition factory is an implementation of:
+### Landing Zone Flight Step Providers
 
+Landing zones are implemented using providers classes which return list of necessary steps to create a landing zone. Each provider should implement following interface `StepsDefinitionProvider`
 ```java
-public interface LandingZoneDefinitionFactory {
-    DefinitionHeader header();
+public interface StepsDefinitionProvider {
 
-    List<DefinitionVersion> availableVersions();
+  String landingZoneDefinition();
 
-    LandingZoneDefinable create(DefinitionVersion version);
-}
-```
+  DefinitionHeader header();
 
-The library includes an abstract class that expects the Azure Resource Manager (ARM)
-clients: `ArmClientsDefinitionFactory`. Factories should extend this class.
+  List<DefinitionVersion> availableVersions();
 
-In addition, factories:
+  List<Pair<Step, RetryRule>> get(
+      ArmManagers armManagers,
+      ParametersResolver parametersResolver,
+      ResourceNameProvider resourceNameProvider,
+      LandingZoneProtectedDataConfiguration landingZoneProtectedDataConfiguration);
+}
+```
 
-- Must be implemented in the `factories` package.
-- Must have a package scoped parameterless constructor.
+### Azure Resource Flight Step
+Each step is responsible to create a certain Azure resource and is represented as a separate class. All steps are inherited from the base class named `BaseResourceCreateStep`.
+All steps are located in the following package `bio.terra.landingzone.stairway.flight.create.resource.step`. If a certain landing zone needs a new resource it is required only to introduce specific step and include it into specific provider.
 
+Below is an example of a step responsible for vnet creation. `createResource` is the central part of the class implementation and it contains the logic responsible for Azure resource creation. 
```

Comment:
I would suggest just linking to the file, as it seems very likely that this will get out-of-date.